lambdas/accesstoken/src/main/java/uk/gov/di/ipv/cri/passport/accesstoken/validation/TokenRequestValidator.java

```diff
@@ -0,0 +1,117 @@
+
+    private String clientId;
+
+    public TokenRequestValidator(ConfigurationService configurationService) {
+        this.configurationService = configurationService;
+        this.verifier = getClientAuthVerifier(configurationService);
+    }
+
+    public void authenticateClient(String requestBody, Map<String, String> queryParams)
+            throws ClientAuthenticationException {
+        if (!queryParams.containsKey(CLIENT_ASSERTION_PARAM)) {
+            if (!queryParams.containsKey(CLIENT_ID_PARAM)) {
+                LOGGER.error(
+                        "Missing either client_assertion or client_id values in request. Failed to establish client_id value.");
+                throw new ClientAuthenticationException(
+                        "Unknown client, no client_id value or client_assertion jwt found in request");
+            } else {
+                clientId = queryParams.get(CLIENT_ID_PARAM);
+
+                String clientAuthenticationMethod = getClientAuthenticationMethod(clientId);
+
+                if (clientAuthenticationMethod.equals(JWT)) {
+                    LOGGER.error("Missing client_assertion jwt for configured client {}", clientId);
+                    throw new ClientAuthenticationException(
+                            String.format(
+                                    "Missing client_assertion jwt for configured client '%s'",
+                                    clientId));
+                }
+            }
+        } else {
+            PrivateKeyJWT clientJwt;
+            try {
+                clientJwt = PrivateKeyJWT.parse(requestBody);
+
+                clientId = clientJwt.getClientID().getValue();
+
+                String clientAuthenticationMethod = getClientAuthenticationMethod(clientId);
+
+                if (clientAuthenticationMethod.equals(NONE)) {
+                    return;
+                }
+
+                verifier.verify(clientJwt, null, null);
+                validateExpiration(clientJwt.getJWTAuthenticationClaimsSet());
+            } catch (ParseException | InvalidClientException | JOSEException e) {
+                LOGGER.error("Validation of client_assertion jwt failed");
+                throw new ClientAuthenticationException(e);
+            }
+        }
+    }
+
+    private String getClientAuthenticationMethod(String clientId)
+            throws ClientAuthenticationException {
+        String clientAuthenticationMethod =
+                configurationService.getClientAuthenticationMethod(clientId);
+        if (clientAuthenticationMethod == null) {
+            LOGGER.error("Failed to find config for client ID: {}", clientId);
+            throw new ClientAuthenticationException(
+                    String.format("Config for client ID '%s' not found", clientId));
+        }
+        return clientAuthenticationMethod;
+    }
+
+    private void validateExpiration(JWTAuthenticationClaimsSet claimsSet)
```

Comment:
Could this be named to be more descriptive about it validating that the expiry isn't too far in the future?